src/signer/src/sign/btc.rs

```diff
@@ -0,0 +1,23 @@
+//! Code for signing Bitcoin transactions.
+use crate::derivation_path;
+use crate::state::read_config;
+use candid::Principal;
+use ic_cdk::api::management_canister::ecdsa::{
+    ecdsa_public_key, EcdsaCurve, EcdsaKeyId, EcdsaPublicKeyArgument,
+};
+
+/// Computes the public key of the specified principal.
+pub async fn ecdsa_pubkey_of(principal: &Principal) -> Vec<u8> {
```

Comment:
I imagine that this function differs from the correspondent ETH one just for the derivation path. Would it be  worth to create a generic function that accepts the derivation path as input too?

Reply:
There will be a generic signing API coming soon, and these Bitcoin and Ethereum operations will be using that instead of the canister API.  So, yes, in a way! :-)